### ISSUE TITLE
prediction difference between batch=1 and batch=16

### ISSUE THREAD

**Issue by alanbekker** [astorfi/3D-convolutional-speaker-recognition#16]
Any ideas why I'm receiving different prediction values when running with batch_size=1,16?
find code below:
Thanks!

  def predict(self,speech_input):
        labels = np.empty(0, int)
        labels = np.append(labels, range(speech_input.shape[0]), axis=0)
        feature,logits,_ = self.session.run(
            [self.features,self.logits,self.end_points_speech],
            feed_dict={self.is_training: False, self.batch_dynamic: labels.shape[0],
                       self.margin_imp_tensor: 50,
                       self.batch_speech: speech_input})
                       #self.batch_labels: labels.reshape([labels.shape[0], 1])})

        # Extracting the associated numpy array.
        #print (feature[0])

        return  feature,logits

**Comment by astorfi** [comment]
Are you running on your own dataset?

**Comment by alanbekker** [comment]
Yes

**Comment by astorfi** [comment]
I do not get your question actually. Why are you expecting to get the same prediction values?

**Comment by alanbekker** [comment]
Because I'm in inference stage after I finished the training stage

look this pseudo code:

for i in batch16:
 yi=model.predict(xi)

and I compare it to:
y=model.predict(batch16)

and this 16 length vector y look different...

and the results look different

**Comment by astorfi** [comment]
What do you mean by batch16?

**Comment by alanbekker** [comment]
Batch16 is a (16,80,40,20) tensor a opposed to yi which is (1,80,40,20)
signle example

**Comment by astorfi** [comment]
I think you should debug the values more carefully. It's weird. It's hard for me to have an idea about the details and the values that you are getting. Please see if the averaging is performing correctly. This might be due to the drastic changes happened in new TensorFlow versions.

**Comment by astorfi** [comment]
Please open this issue if the problem has not been resolved!
Thanks
